internal/security/pipedhexreader/pipedhexreader.go

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package pipedhexreader
+
+import (
+	"bufio"
+	"encoding/hex"
+	"os/exec"
+	"strings"
+)
+
+// pipedHexReader stores instance data for the pipedhexreader
+type pipedHexReader struct{}
+
+// NewPipedHexReader creates a new PipedHexReader
+func NewPipedHexReader() PipedHexReader {
+	return &pipedHexReader{}
+}
+
+// ReadHexBytesFromExe see interface.go
+func (phr *pipedHexReader) ReadHexBytesFromExe(executablePath string, args []string) ([]byte, error) {
+	sanitizedExecutable, err := exec.LookPath(executablePath)
```

Comment:
Wouldn't requiring a fully qualified path lock things down a bit more? I suppose the only the only thing leaked would be the set of command-line args passed to the IKM application...

Reply:
LookPath("echo") and LookPath("/bin/echo") both return /bin/echo.   I think the current implementation preserves flexibility for a user that wants to lock it down to a full path.